### ISSUE TITLE
cannot reproduce the results on human3.6m

### ISSUE THREAD

**Issue by a2394797795** [Jeff-sjtu/res-loglikelihood-regression#8]
Hi, 
I have tried to train the model on human3.6m dataset, but cannot get the same performance as the provided pretrained model 'h36m-laplace-rle.pth'. 
I was using the annotation files and codes from this repository, and the environment is based on pytorch1.8.

The training behaives normally at the beginning, but after some epoch, the loss value was increased and MPJPE reached 200. 
```
Train-9 epoch | loss:-179.81700464 | acc:0.5870
##### Epoch 9 | gt results: 78.48804327541252/78.48804327541252 #####
############# Starting Epoch 10 | LR: 0.001 #############
Train-10 epoch | loss:-178.24089413 | acc:0.5794
############# Starting Epoch 11 | LR: 0.001 #############
Train-11 epoch | loss:-105.90890308 | acc:0.2606
##### Epoch 11 | gt results: 206.2949851515231/78.48804327541252 #####
############# Starting Epoch 12 | LR: 0.001 #############
Train-12 epoch | loss:-99.68143688 | acc:0.2141
############# Starting Epoch 13 | LR: 0.001 #############
Train-13 epoch | loss:-134.96754921 | acc:0.3854
```

Before lr step, the model would collapse. 
```
############# Starting Epoch 60 | LR: 0.001 #############
Train-60 epoch | loss:-99.61992546 | acc:0.2145
##### Epoch 60 | gt results: 183.75314692309206/71.80444962169011 #####
############# Starting Epoch 61 | LR: 0.001 #############
Train-61 epoch | loss:16064.89554501 | acc:0.0533
##### Epoch 61 | gt results: 973.8534021229914/71.80444962169011 #####
############# Starting Epoch 62 | LR: 0.001 #############
Train-62 epoch | loss:-18.63251462 | acc:0.0071
```
 
I doubted if it was from the very large lr, and then experimented with initial lr 10e-4. The loss value was lower but also increased after some epoch.
```
##### Epoch 64 | gt results: 83.14857669377162/64.38682879287299 #####
############# Starting Epoch 65 | LR: 0.0001 #############
Train-65 epoch | loss:-208.61684037 | acc:0.6639
##### Epoch 65 | gt results: 74.92741776420186/64.38682879287299 #####
############# Starting Epoch 66 | LR: 0.0001 #############
Train-66 epoch | loss:-207.51670638 | acc:0.6620
##### Epoch 66 | gt results: 77.90103486437523/64.38682879287299 #####
############# Starting Epoch 67 | LR: 0.0001 #############
Train-67 epoch | loss:-176.32472946 | acc:0.5414
##### Epoch 67 | gt results: 204.0743792529304/64.38682879287299 #####
############# Starting Epoch 68 | LR: 0.0001 #############
Train-68 epoch | loss:-126.63842224 | acc:0.3175
##### Epoch 68 | gt results: 124.20229629684249/64.38682879287299 #####
```

If the lr is dropped to 1e-5 before the , I got a final result of 66 MPJPE, compared to the 38 MPJPE by pretrained model.

Additionally, I have tried to downgrade the pytorch to 1.5, but got similar phenomenon.

Do you have any suggestions? Thanks a lot for your help.

**Comment by Jeff-sjtu** [comment]
Hi @a2394797795, I am trying to reproducing this bug. I will provide the solution as soon as possible.

**Comment by jeffffffli** [comment]
Hi @a2394797795,

We have fixed the bug. Now the model can be trained normally. The bug is caused by [this line](https://github.com/Jeff-sjtu/res-loglikelihood-regression/blob/master/rlepose/models/regression_nf_3d.py#L153). We accidentally subtract the root depth during training. Therefore, `bar_mu` of root joints will always have zeros `z`, which increase the probability of zero and the normalizing flow model will learn a wrong distribution.

**Comment by a2394797795** [comment]
Thanks for your reply! I will close this issue.
